### PR TITLE
metrics: storing metrics workflow artifacts

### PR DESCRIPTION
**Pull request by dborquez**
This PR enables storing metrics workflow artifacts in two separated flavours: clh and qemu.

Fixes: #7239

### CODE REVIEW

File: .github/workflows/run-metrics.yaml [hunk 1]
```diff
@@ -40,3 +40,13 @@ jobs:
       - name: run memory usage inside container test
         run:  bash tests/metrics/gha-run.sh run-test-memory-usage-inside-container
 
+      - name: Make metrics tarball ${{ matrix.vmm }}
```
Comment (GabyCT) [inline .github/workflows/run-metrics.yaml:43]: should we use lowercase in all the names to have uniformity?

Comment (dborquez) [reply .github/workflows/run-metrics.yaml]: @GabyCT change applied, thanks

**Review by fidencio — APPROVED** [review]
lgtm, thanks @dborquez!

**Review by GabyCT — APPROVED** [review]

**Review by amshinde — APPROVED** [review]

**Pull request merged**